osu.Game/Screens/Play/GameplayState.cs

```diff
@@ -30,7 +31,12 @@ public class GameplayState
         /// <summary>
         /// The mods applied to the gameplay.
         /// </summary>
-        public IReadOnlyList<Mod> Mods;
+        public readonly IReadOnlyList<Mod> Mods;
+
+        /// <summary>
+        /// The gameplay score.
+        /// </summary>
+        public Score? Score { get; set; }
```

Comment:
I'd really hope this is `readonly`, are you okay with me changing the flow in player to create the score early enough to change that? Seems like it should be possible.

Reply:
Make sure to test it. Only reason I didn’t is because I didn’t want to look into how ReplayPlayer / Autoplay / Solo handle things…

Reply:
Yep. On a quick look they don't seem like they'd be affected but will test.